haystack/finder.py

```diff
@@ -29,6 +29,12 @@ def get_answers(self, question: str, top_k_reader: int = 1, top_k_retriever: int
         :return:
         """
 
+        if self.reader is None:
```

Comment:
It might be useful to allow a Finder with only a Reader or a Retriever. There's already an implementation [here](https://github.com/deepset-ai/haystack/blob/bf8e506c45a9c7084e5f122d0ada706ac070e1e3/haystack/api/controller/search.py#L58) in the APIs where a Finder can be used without Reader for FAQ matching.